app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class UsersController < API::V1::ApplicationController
+      include API::V1::ErrorHandlerConcern
+
+      skip_before_action :doorkeeper_authorize!, only: :create
+
+      def create
+        user = User.new(create_params.except(:client_id))
+        client_app = Doorkeeper::Application.find_by(uid: create_params[:client_id])
```

Comment:
can this client app part be moved to a before callback, then you wont need to use return as well. 
```
   before_action :ensure_valid_client, only: :create
 
   def ensure_valid_client
     @client_app = Doorkeeper::Application.find_by(uid: create_params[:client_id])

     render_error 'Invalid client ID', status: :forbidden, source: :client_id if @client_app.blank?
   end
end

Reply:
Yes we can (as we only need the client id inside the create action, so we can get it from params again).
But is it really worth it? As we do not expect to have more actions in this controller ; isn't it trying to solve a problem that is not here? (I'm asking as I probably don't know about something, so I want to be sure to understand the why -- in order to not reproduce the same mistake twice)
Thanks!! :)

Reply:
@malparty Firstly, our action create would become shorter and look much cleaner. Secondly, this is something which can be reusable across other actions in future. For instance, have a look at doorkeeper tokens controller https://github.com/doorkeeper-gem/doorkeeper/blob/main/app/controllers/doorkeeper/tokens_controller.rb. In that, they do something similar

```
    before_action :validate_presence_of_client, only: [:revoke]
```

Reply:
Thanks a lot! I could not see it clearly before working on the serializer, but not the "before/after" of the create method is impressive! Much clearer! :)